universal-login-wallet/src/ui/react/App.tsx

```diff
@@ -19,7 +19,18 @@ import {ConnectionNotificationScreen} from './ConnectAccount/ConnectionNotificat
 const App = () => {
   const {history} = useRouter();
   const modalService = createModalService<WalletModalType, void>();
-  const {walletService, sdk} = useServices();
+  const {walletService, walletStorageService, sdk} = useServices();
+  const [appReady, setAppReady] = useState(false);
+
+  useLayoutEffect(() => {
+    walletStorageService.load();
+    setAppReady(true);
+  }, []);
+
+  if (!appReady) {
+    return null;
```

Comment:
Shall we do `return 'Loading...';` instead?

Reply:
This will result in a flash of text in this case. I've only added it here since the wallet service is queried directly, which means that it wouldn't update the component by itself. So we need a state just to trigger a re-render.

Reply:
We should think about a way to add an ability to watch for changes in the wallet service.

Reply:
It is on the roadmap.